src/foam/nanos/ruler/RulerDAO.js

```diff
@@ -123,14 +172,94 @@
         }
       ],
       javaCode: `
-
       for ( Object key : sink.getGroupKeys() ) {
         List<Rule> group = ((ArraySink) sink.getGroups().get(key)).getArray();
         if ( ! group.isEmpty() ) {
+          Collections.sort(group, Rule.PRIORITY);
+          Collections.reverse(group);
```

Comment:
You can combine these two in a more efficient way by doing either:
Collections.sort(group, new foam.mlang.order.Desc(Rule.PRIORITY));
or
Collections.sort(group, DESC(Rule.PRIORITY)); // if you've statically imported mlang and pulled on foam2